docs/articles/documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.md

```diff
@@ -0,0 +1,445 @@
+    path: filePath | module: moduleName | content: code
+})
+
+fixture.clientScripts({
+    page: url,
+    path: filePath | module: moduleName | content: code
+}, ...)
+
+fixture.clientScripts([
+    {
+        page: url,
+        path: filePath | module: moduleName | content: code
+    }
+])
+```
+
+```text
+test.clientScripts({
+    page: url,
+    path: filePath | module: moduleName | content: code
+})
+
+test.clientScripts({
+    page: url,
+    path: filePath | module: moduleName | content: code
+}, ...)
+
+test.clientScripts([
+    {
+        page: url,
+        path: filePath | module: moduleName | content: code
+    }
+])
+```
+
+Property  | Type                | Description
+--------- | ------------------- | ---------------------------------------------------------------------------
+`url`    | String &#124; RegExp | Specify a page URL to add scripts to a single page, or a regular expression to add scripts to pages whose URLs match this expression. Regular expressions are not supported in the [clientScripts](../configuration-file.md#clientscripts) configuration file property.
+
+**Examples**
+
+```js
+runner.clientScripts({
+    page: /\/user\/profile\//,
+    path: 'dist/jquery.js'
+});
+```
+
+```json
+{
+    "clientScripts": {
+        "page": "https://myapp.com/page/",
+        "content": "Geolocation.prototype.getCurrentPosition = () => new Positon(0, 0);"
+    }
+}
+```
+
+> If the target page redirects to a different URL, ensure that the `page` property matches the destination URL. Otherwise, scripts are not injected.
+
+### Specify Multiple Scripts
+
+You can pass multiple arguments or an array to the `clientScripts` methods:
+
+```js
+fixture `My fixture`
+    .page `https://example.com`
+    .clientScripts('scripts/react-helpers.js', { content: 'Date.prototype.getTime = () => 42;' });
+```
+
+```js
+runner.clientScripts(['scripts/react-helpers.js', 'dist/jquery.js']);
+```
+
+The [clientScripts](../configuration-file.md#clientscripts) configuration file property can also take arrays:
+
+```json
+{
+    "clientScripts": ["vue-helpers.js", {
+        "page": "https://mycorp.com/login/",
+        "module": "lodash"
+    }]
+}
+```
+
+The [--cs (--client-scripts)](../command-line-interface.md#--cs-pathpath2---client-scripts-pathpath2) command line option supports multiple arguments as well:
+
+```sh
+testcafe chrome test.js --client-scripts=mockDate.js,assets/react-helpers.js
+```
+
+Note that the `page`, `content` and `module` properties cannot take arrays. To inject multiple scripts into the same page, pass one argument for each script.
+
+```js
+const scripts = ['test1.js', 'test2.js', 'test3.js'];
+
+runner.clientScripts(scripts.map(script => {
+    path: script,
+    page: 'http://example.com'
+}));
+```
+
+## Access DOM in the Injected Scripts
+
+TestCafe injects scripts into the `head` tag, which means they are executed before DOM is loaded.
+
+To access DOM in these scripts, wait until the `DOMContentLoaded` event fires:
```

Comment:
To access the DOM in these scripts, wait until the `DOMContentLoaded` event fires: